lib/internal/modules/cjs/loader.js

```diff
@@ -297,8 +327,58 @@ function findLongestRegisteredExtension(filename) {
   return '.js';
 }
 
+function resolveExports(nmPath, request, absoluteRequest) {
+  // This only applies to requests of a specific form:
+  // 1. (name)/.*
+  // 2. @(scope)/(name)/.*
+  // The implementation's behavior is meant to mirror resolution in ESM.
+  if (experimentalExports && !absoluteRequest) {
+    // TODO(jkrems): Support folder exports ("pkg-name/dir/")
+    const [, name, expansion] =
+      request.match(/^((?:@[^./@][^/@]*\/)?[^@./][^/]*)(\/.*)$/) || [];
```

Comment:
We should possibly include `\` in the exclusion groups, although no need to support `\` as a separator I think.

Reply:
Good point, added `\` to all the exclusions. Also one more thing to compare to the ESM implementation.